amethyst_core/src/dispatcher.rs

```diff
@@ -50,19 +62,31 @@ impl DispatcherData {
 /// A builder which is used to construct [Dispatcher] from multiple systems and system bundles.
 #[derive(Default)]
 pub struct DispatcherBuilder {
-    items: Vec<DispatcherItem>,
+    items: Vec<DispatcherItem<'static>>,
 }
 
-impl DispatcherBuilder {
+impl<'a> DispatcherBuilder {
     /// Adds a system to the schedule.
-    pub fn add_system<T: ParallelRunnable + 'static>(&mut self, system: T) -> &mut Self {
-        self.items.push(DispatcherItem::System(Box::new(system)));
+    pub fn add_system<S: System<'static> + 'static>(&mut self, system: Box<S>) -> &mut Self {
+        let s: &'static mut S = Box::leak(system);
```

Comment:
That's I think the most important change here. I think that, as systems are meant to live during all the lifetime of the app, it make sense.

Reply:
> 
> 
> @ezpuzz This is a huge diff. Can you please go into more detail about:
> 
>     * What this change does
> 
>     * Why you are making the change
> 
>     * Pros / Cons of this approach
> 
> 
> That would help me immensely in my quest to either review or find a reviewer for this PR. 😉

@CleanCut I've updated the description.